src/daemon/main.cpp

```diff
@@ -12,33 +12,46 @@
 #include <clocale>
 #include <cstdlib>
 #include <iostream>
+#include <memory>
 
 #include "common/version.h"
 #include "config.h"
 #include "daemon/arguments.h"
+#include "daemon/backends/connman_backend.h"
 #include "daemon/daemon.h"
 
+namespace
+{
+    std::unique_ptr<ConnectivityManager::Daemon::Backend> create_backend()
+    {
+#if CONNECTIVITY_MANAGER_BACKEND == CONNECTIVITY_MANAGER_BACKEND_CONNMAN
+        return std::make_unique<ConnectivityManager::Daemon::ConnManBackend>();
+#else
+#    error "Mising backend in create_backend()."
+#endif
+    }
+}
+
 int main(int argc, char *argv[])
 {
     std::setlocale(LC_ALL, "");
 
     Glib::init();
     Gio::init();
 
-    auto arguments = TemplateDBusService::Daemon::Arguments::parse(argc, argv, std::cout);
+    auto arguments = ConnectivityManager::Daemon::Arguments::parse(argc, argv, std::cout);
     if (!arguments)
         return EXIT_FAILURE;
 
     if (arguments->print_version_and_exit) {
-        std::cout << Glib::get_prgname() << " " << TemplateDBusService::Common::VERSION << '\n';
+        std::cout << Glib::get_prgname() << " " << ConnectivityManager::Common::VERSION << '\n';
         return EXIT_SUCCESS;
     }
 
-#ifdef TEMPLATE_DBUS_SERVICE_FOO
-    g_message("Using feature FOO. REMOVE ME!");
-#endif
+    ConnectivityManager::Daemon::Daemon daemon(create_backend());
 
-    TemplateDBusService::Daemon::Daemon daemon;
+    if (!daemon.run())
+        return EXIT_FAILURE;
 
-    return daemon.run();
```

Comment:
Seems like I forgot to sync over this change (making run() return exit code instead of bool) @thenor and me did when extracting code from connectivity-manager to the template. Only noticed now when I rebased cm changes to the template. Will change to return exit code in cm as well.